pushy/src/main/java/com/eatthepath/pushy/apns/util/LiveActivityApnsPayloadBuilder.java

```diff
@@ -0,0 +1,257 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.eatthepath.pushy.apns.util;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * <p>A base utility class for constructing JSON payloads suitable for inclusion in APNs push notifications for Live
+ * Activities. Payload builders are reusable, but are <em>not</em> thread-safe.</p>
+ *
+ * @see <a href="https://developer.apple.com/documentation/activitykit/update-and-end-your-live-activity-with-remote-push-notifications">Updating and ending your Live Activity with remote push notifications</a>
+ *
+ * @since 0.15.2
+ */
+@SuppressWarnings({"UnusedReturnValue", "unused"})
+public abstract class LiveActivityApnsPayloadBuilder {
+
+    private String alertBody = null;
+    private String alertTitle = null;
+    private String soundFileName = null;
+    private LiveActivityEvent event = null;
+    private Instant timestamp = null;
+    private Instant dismissalDate = null;
+    private final HashMap<String, Object> contentState = new HashMap<>();
+
+
+    private static final String APS_KEY = "aps";
+    private static final String ALERT_KEY = "alert";
+    private static final String SOUND_KEY = "sound";
+    private static final String TIMESTAMP_KEY = "timestamp";
+    private static final String DISMISSAL_DATE_KEY = "dismissal-date";
+    private static final String EVENT_KEY = "event";
+    private static final String CONTENT_STATE_KEY = "content-state";
+    private static final String ALERT_TITLE_KEY = "title";
+    private static final String ALERT_SUBTITLE_KEY = "subtitle";
+    private static final String ALERT_BODY_KEY = "body";
+
+    public static final String[] EMPTY_STRING_ARRAY = new String[0];
+
+    /**
+     * The name of the iOS default push notification sound.
+     *
+     * @see LiveActivityApnsPayloadBuilder#setAlertSound(String)
+     */
+    public static final String DEFAULT_SOUND_FILENAME = "default";
+
+    /**
+     * Sets the literal text of the alert message. The expanded view of the Live Activity will be shown for a few
+     * seconds if an alert body and alert title are sent. It is unclear if the actual text is used anywhere.
+     *
+     * @param alertBody the literal message to be shown for this push notification
+     *
+     * @return a reference to this payload builder
+     */
+    public LiveActivityApnsPayloadBuilder setAlertBody(final String alertBody) {
```

Comment:
This will need to support localization too: https://developer.apple.com/documentation/activitykit/update-and-end-your-live-activity-with-remote-push-notifications

Reply:
You're right, fixed!